spec/services/schools/school_metrics_generator_service_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'rails_helper'
+
+describe Schools::SchoolMetricsGeneratorService, type: :service do
+
+  let!(:benchmark_run)    { BenchmarkResultGenerationRun.create! }
+
+  let(:school)            { create(:school) }
+  #this will create an empty meter collection as the school has no data
+  let(:meter_collection)  { Amr::AnalyticsMeterCollectionFactory.new(school).validated }
+
+  subject(:service)       { Schools::SchoolMetricsGeneratorService.new(school: school, meter_collection: meter_collection)}
+
+  let(:stub)              { double('service_stub') }
+
+  describe '#perform' do
+
+    context 'it should update school configuration' do
```

Comment:
I appreciate this spec is taken from elsewhere and understand you might not have had time to change it. 

Context labels - I'd be labelling them more like "updating school configuration" or "when updating school configuration" and I'd be looking to put the expect statements also inside the 'it' block (and not the 'before' block with them being tests and not test setup).

Reply:
Yes, good shout I rushed this a bit. I've reworked the specs as suggested.